### PR TITLE
[W5.11][T13-1]Lu Han

### PR DESCRIPTION
**Pull request by luhan02**

### CODE REVIEW

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 1]
```diff
@@ -1,11 +1,6 @@
 package seedu.addressbook.data.person;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Iterator;
-import java.util.List;
+import java.util.*;
```
Comment (QzSG) [inline src/seedu/addressbook/data/person/UniquePersonList.java:3]: Configure IntelliJ to not auto collapse imports by doing
`Preferences` > `Editor` > `Code Style` > `Java` > `Imports` tab set `Class count to use import with '*'`

File: src/seedu/addressbook/commands/SortCommand.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
```
Comment (QzSG) [inline src/seedu/addressbook/commands/SortCommand.java:7]: https://github.com/nusCS2113-AY1819S1/addressbook-level2/blob/119927019e3cd76101271cfb7ea63a74bc309f29/src/seedu/addressbook/commands/SortCommand.java#L5-L8

You forgot to remove unused imports, I think only `import java.util.List` is required form `java.util`?

File: docs/UserGuide.adoc [hunk 3]
```diff
@@ -72,21 +72,26 @@ Examples:
...
+
 == Finding all persons containing any keyword in their name: `find`
 
 Finds persons whose names contain any of the given keywords. +
 Format: `find KEYWORD [MORE_KEYWORDS]`
 
 [NOTE]
 ====
-The search is case sensitive, the order of the keywords does not matter, only the name is searched,
+The search is case ininsensitive, the order of the keywords does not matter, only the name is searched,
```
Comment (ewaldhew) [inline docs/UserGuide.adoc:87]: Woops, a typo here! 😂

File: src/seedu/addressbook/commands/FindCommand.java [hunk 4]
```diff
@@ -50,8 +50,18 @@ public CommandResult execute() {
         final List<ReadOnlyPerson> matchedPersons = new ArrayList<>();
         for (ReadOnlyPerson person : addressBook.getAllPersons()) {
             final Set<String> wordsInName = new HashSet<>(person.getName().getWordsInName());
-            if (!Collections.disjoint(wordsInName, keywords)) {
-                matchedPersons.add(person);
+//            if (!Collections.disjoint(wordsInName, keywords)) {
+//                matchedPersons.add(person);
+//            }
```
Comment (ewaldhew) [inline src/seedu/addressbook/commands/FindCommand.java:55]: Try not to commit commented-out code like this. Git keeps a record of code changes, if you really need it back in the future, you can just revert.

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 5]
```diff
@@ -81,6 +76,20 @@ public UniquePersonList(UniquePersonList source) {
         return Collections.unmodifiableList(internalList);
     }
 
+    private void Sorting() {
```
Comment (ewaldhew) [inline src/seedu/addressbook/data/person/UniquePersonList.java:79]: Missing header comments, and naming does not follow coding standard. Names should be in camelCase.

File: src/seedu/addressbook/data/person/UniquePersonList.java [hunk 6]
```diff
@@ -81,6 +76,20 @@ public UniquePersonList(UniquePersonList source) {
...
+
+        Collections.sort(internalList, new Comparator<Person>() {
+            public int compare(Person p1 , Person p2) {
+                return p1.getName().fullName.compareTo(p2.getName().toString()); //return string
+            }
+        });
+    }
+
+    public List<ReadOnlyPerson> sortedList() {
+                Sorting();
```
Comment (ewaldhew) [inline src/seedu/addressbook/data/person/UniquePersonList.java:89]: Indentation's too far in, we use only 4 spaces per level

File: test/expected.txt [hunk 7]
```diff
@@ -86,66 +88,126 @@
...
+|| Enter command: || [Command entered:  add Esther Potato p/555555 e/esther@not.a.real.potato pa/555, epsilon street t/tubers t/starchy]
+|| This person already exists in the address book
+|| ===================================================
+|| Enter command: || [Command entered:  sort]
+|| 	1. Adam Brown Phone: 111111 Email: adam@gmail.com Address: 111, alpha street Tags:
+|| 	2. Betsy Choo Tags: [secretive]
+|| 	3. Charlie Dickson Email: charlie.d@nus.edu.sg Address: 333, gamma street Tags: [school][friends]
+|| 	4. Dickson Ee Phone: 444444 Address: 444, delta street Tags: [friends]
+|| 	5. Esther Potato Phone: 555555 Email: esther@not.a.real.potato Tags: [tubers][starchy]
+||
```
Comment (ewaldhew) [inline test/expected.txt:197]: Good work updating tests, but it would be better if you sorted a list that's not already sorted, so that you can be sure the `sort` command is _really_ working

File: test/expected.txt [hunk 8]
```diff
@@ -264,42 +335,42 @@
...
 || 0 persons listed!
 || ===================================================
 || Enter command: || [Command entered:  exit]
 || Exiting Address Book as requested ...
 || ===================================================
 || Good bye!
 || ===================================================
-|| ===================================================
+|| ===================================================
```
Comment (ewaldhew) [inline test/expected.txt:376]: A lot of whitespace changes. These changes not only cause your tests to fail, they are also unrelated changes that "pollute" your PR 😢

**Review by ewaldhew — COMMENTED** [review]
Good effort, mostly coding standard mistakes so do watch out for these next time. Close PR after reading
